### PR TITLE
Replace `Pimcore\Cache\Runtime` inside `models/*` with `Pimcore\Cache\RuntimeCacheTrait`

### LINKED ISSUE #11339: [Improvement]: Use `RuntimeCacheTrait` whenever possible instead of dealing with `Cache\Runtime` directly
### Improvement description

`RuntimeCacheTrait` was introduced by #11254 

There are multiple places in the code where `Pimcore\Cache\Runtime` is used directly in models, this should be changed to the new trait.

### PR DESCRIPTION
**Pull request by aashan10**
## Changes in this pull request  
Resolves #11339

### CODE REVIEW

File: models/Asset.php [hunk 1]
```diff
@@ -248,7 +251,7 @@ public static function getById($id, $force = false)
                     $asset->getDao()->getById($id);
                 }
 
-                \Pimcore\Cache\Runtime::set($cacheKey, $asset);
+                self::setCache($asset, $cacheKey);
```
Comment (jdreesen) [inline models/Asset.php:254]: The (non-runtime-)cache is now saved twice, no? `RuntimeCacheTrait::setCache()` saves both caches, and a few lines below (259) the non-runtime-cache is saved again.

File: models/Asset.php [hunk 2]
```diff
@@ -258,7 +261,7 @@ public static function getById($id, $force = false)
                 return null;
             }
         } else {
-            \Pimcore\Cache\Runtime::set($cacheKey, $asset);
+            self::setCache($asset, $cacheKey);
```
Comment (jdreesen) [inline models/Asset.php:264]: This now saves the non-runtime-cache, too, which wasn't the case before. Is this correct/intended?

Comment (aashan10) [reply models/Asset.php:264]: No, it is not intended. This is still a work in progress.

**Comment by aashan10** [comment]
Closing this pull request as discussed [here](https://github.com/pimcore/pimcore/issues/11339#issuecomment-1151152660)

**Pull request closed without merging**